jobs/base_schedulers.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package jobs
+
+import (
+	"time"
+
+	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/mattermost/mattermost-server/v6/shared/mlog"
+)
+
+type PeriodicScheduler struct {
+	jobs        *JobServer
+	minutes     int
+	jobType     string
+	enabledFunc func(cfg *model.Config) bool
+}
+
+func NewPeridicScheduler(jobs *JobServer, jobType string, minutes int, enabledFunc func(cfg *model.Config) bool) *PeriodicScheduler {
+	return &PeriodicScheduler{
+		minutes:     minutes,
+		jobType:     jobType,
+		enabledFunc: enabledFunc,
+		jobs:        jobs,
+	}
+}
+
+func (scheduler *PeriodicScheduler) Enabled(cfg *model.Config) bool {
+	return scheduler.enabledFunc(cfg)
+}
+
+func (scheduler *PeriodicScheduler) NextScheduleTime(_ *model.Config, _ time.Time, _ bool, _ *model.Job) *time.Time {
+	nextTime := time.Now().Add(time.Duration(scheduler.minutes) * time.Minute)
+	return &nextTime
+}
+
+func (scheduler *PeriodicScheduler) ScheduleJob(_ *model.Config, _ bool, _ *model.Job) (*model.Job, *model.AppError) {
+	return scheduler.jobs.CreateJob(scheduler.jobType, nil)
+}
+
+type DailyScheduler struct {
+	jobs        *JobServer
+	timeConfig  func(cfg *model.Config) string
```

Comment:
It is not strictly that, I was getting the config setting for that, and then letting the parsing process to the daily scheduler, moving this to the scheduler leads to certain code repetition, I did that, because I think is less leaky abstraction, and probably we can do better in a subsequent iteration.

Reply:
Done